code/hsec-tools/src/Security/Advisories/Generate/Snapshot.hs

```diff
@@ -0,0 +1,76 @@
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Security.Advisories.Generate.Snapshot
+  ( createSnapshot,
+  )
+where
+
+import Data.Aeson (ToJSON, encodeFile)
+import Data.Default (def)
+import qualified Data.Text.IO as T
+import Data.Time (UTCTime, zonedTimeToUTC)
+import Data.Version (Version)
+import GHC.Generics (Generic)
+import Paths_hsec_tools (version)
+import Security.Advisories.Core.Advisory
+import Security.Advisories.Filesystem (advisoryFromFile, forAdvisory, forReserved)
+import Security.Advisories.Format (codecFrontMatter, fromAdvisory)
+import System.Directory (copyFileWithMetadata, createDirectoryIfMissing)
+import System.FilePath (takeDirectory, (</>))
+import System.IO (hPrint, stderr)
+import Text.Pandoc (Block (CodeBlock), Pandoc (Pandoc), nullMeta, runIOorExplode)
+import Text.Pandoc.Writers (writeCommonMark)
+import qualified Toml
+import Validation (Validation (..))
+
+-- * Actions
+
+createSnapshot :: FilePath -> FilePath -> IO ()
+createSnapshot src dst = do
+  let toDstFilePath orig = dst </> drop (length src + 1) orig
+
+  forReserved src $ \p _ -> do
+    createDirectoryIfMissing True $ takeDirectory $ toDstFilePath p
+    putStrLn $ "Copying '" <> p <> "' to '" <> toDstFilePath p <> "'"
```

Comment:
`hputStrLn sterror`

Reply:
It's not an error

Reply:
I think it’s a fair point that hasufell made, that things that are not supposed to be parsed but are just informative output go to stderr.

Reply:
I agree.  Diagnostic output should go to stderr.

Reply:
There are other cases below also... but I don't see this as a show-stopper.